### DISCUSSION TITLE
左側にUSBを接続したときしか正しく動作しません

### DISCUSSION THREAD

**Discussion by zinfyk** [Yowkees/keyball#579] [Q&A]
左側にUSB接続すると左右ともに正しく動作しますが、
右側にUSBを接続すると左側が動作しません。
具体的には、左側はLEDが点灯するだけで、入力を受け付けません。
この時、左側のOLEDは基本的に消灯していますが、ときよりバグったような表示が行われる時があります。
よろしくお願いします。

**Comment by Ryo624** [comment]
ご連絡ありがとうございます。

症状から左右の通信が上手くできていないようですので、TRRSジャックのはんだ付けをやり直していただくことと、もし予備のものがあれば左右をつなぐTRRSケーブルも変えてみると通信がどのようになるか確認していただけますと幸いです。

**Reply by zinfyk** [reply]
ご返信ありがとうございました。
まず、TRRSケーブルが断線していないかをテスターを使いチェックし、問題が無いことを確認しました。
ハンダを見直す前にTRRSケーブルを繋いだ状態で、双方の端子が接続できているかをテスターを使い測りました。
それぞれ問題なく接続されているようでしたが、念の為ハンダをやり直しました。
しかし結果は変わりませんでした。
左側をPCと接続すれば双方問題く動作することを考えると、TRRS部が原因ではないように思いました。

次に、双方のMicroProを入れ替えてみたところ、右側を入れ替えてみたPCと接続したところ正しく動作し、左側をPCと接続すると問題が生じました。
おそらくはMicroProに問題があるように思い、コンスルーがMicroProと正しくハンダ出来ているかをテスターで測りましたが、こちおらも問題はありませんでした。
おそらくは遊舎工房様で購入したMicroProに問題があったものと思われます。

従いまして、Yowkees様のサポートは一旦不要であることをお知らせいたします。
ありがとうございました。

**Comment by zinfyk** [comment]
その後MicroProを交換しTRRSケーブルも予備を用意しました。
現象はあまり解決していません。
PCと接続をすると一時的に正しく利用できますが、しばらくするとPC接続をしていない側は無反応となります。
正しく動作している間に全てのキーがタイプ出来ること、テスト用のファームウエアにてLEDの点滅が行えていることを確認しています。

改めてTRRSジャック同士をテスターで測ったところ、通電が不安定（接触が不安定）であることが確認できました。
２本のTRRSケーブルを使い確認しましたが、両方で同じ現象がおきているようです。
前回ジャックのハンダをやり直すようにアドバイスいただきましたが、ルーペを使い適切にハンダ出来ていることは確認しています。

こちらどのようにしたらよいでしょうか？
よろしくお願いします。

**Reply by Ryo624** [reply]
一度はんだ付けを確認させていただいてもよろしいでしょうか。

TRRSジャックとProMicroのはんだ付けを行った部分の写真を添付していただけますと幸いです。

**Comment by zinfyk** [comment]
こちらとなります。
コンスルーを使っておりますので、MicroPro側と基板裏側の写真も追加させていただきました。
お手数おかけしますが、ご確認いただけますようよろしくお願いいたします。

![PXL_20240517_065613339 MP のコピー](https://github.com/Yowkees/keyball/assets/169494357/dc35862b-6d10-41e3-9062-3ef3aec72a07)
![PXL_20240517_065631979 のコピー](https://github.com/Yowkees/keyball/assets/169494357/c2c72d8e-09cc-4f7d-a5ce-13ebb02dfc67)
![PXL_20240517_065340730 のコピー](https://github.com/Yowkees/keyball/assets/169494357/646da04b-32dd-4c95-8949-83940a642938)
![PXL_20240517_065223386 のコピー](https://github.com/Yowkees/keyball/assets/169494357/1323ccf7-84cd-42b6-99da-e6d2bad72f41)
![PXL_20240517_070639793 MP](https://github.com/Yowkees/keyball/assets/169494357/9cd20474-37b3-443b-978f-10fe38c603c7)
![PXL_20240517_070603168](https://github.com/Yowkees/keyball/assets/169494357/735becec-57d6-4e60-a034-ab342f4d3963)

**Comment by zinfyk** [comment]
補足となります
遊舎工房の購入時には2mmのコンスルーがリコメンドされます。
L側のMicroProには2mmのコンスルーを取り付けていますが、昨日購入したあたらしいMicroPro（R側）には2.5mmのコンスルーを取り付けています。
ひょっとしたら2mmのコンスルーでは短いなどあるのでしょうか？

**Comment by Ryo624** [comment]
ルーペを使ってしっかりはんだ付けを確認していただいたところ申し訳ありませんが、TRRSジャックのはんだ付けをもう一度やり直してみていただいてもよろしいでしょうか。
銀色のパッドにしっかりと馴染むようにしてみてください。
